rust/sbp/src/json/mod.rs

```diff
@@ -123,18 +124,49 @@ impl Formatter for HaskellishFloatFormatter {
 /// All possible error types when encoding/decoding a stream of json messages.
 #[derive(Debug)]
 pub enum JsonError {
-    PayloadParseError(crate::wire_format::PayloadParseError),
+    SbpMsgParseError(crate::messages::SbpMsgParseError),
     CrcError(crate::de::CrcError),
     WriteFrameError(crate::ser::WriteFrameError),
     IoError(io::Error),
     Base64Error(base64::DecodeError),
     SerdeJsonError(serde_json::Error),
 }
 
+impl HandleParseError<Sbp> for JsonError {
+    fn handle_parse_error(self) -> Sbp {
+        match self {
+            Self::SbpMsgParseError(e) => Invalid::from(e).into(),
+            Self::CrcError(e) => Invalid::from(e).into(),
+            Self::WriteFrameError(e) => {
+                panic!("Unable to recover from WriteFrameError {}", e)
+            }
+            Self::IoError(e) => {
+                panic!("Unable to recover from IoError {}", e)
+            }
+
+            Self::Base64Error(e) => {
+                panic!("Unable to recover from Base64 decoding error {}", e)
+            }
+
+            Self::SerdeJsonError(e) => {
+                // We may need to handle this error path if we want to recover
+                // from errors with Json2Json
+                panic!("Unable to recover from SerdeJsonError {}", e)
```

Comment:
FWIW, the original use case of `json2json` was to expand "elided" JSON data that was output from the old Swift Console when it was logging (the data output only included the base64 message payload and a limited number of common fields, this was done because it was too slow to dump the message and it's constituent fields).  The modern console is fast enough that it's able to dump the "unelided" form of JSON.